### PR TITLE
Feature : Allow resizing of files before downloading

### PR DESCRIPTION
**Pull request by biximilien**
I had needed to resize the zipped files to specific formats before downloading them.
This might be useful to someone else.

### CODE REVIEW

File: lib/active_storage/send_zip_helper.rb [hunk 1]
```diff
@@ -13,15 +13,15 @@ module SendZipHelper
     #
     # @param files [ActiveStorage::Attached::One|ActiveStorage::Attached::Many|Array|Hash] file(s) to save
     # @return [String] folder path of saved files
-    def self.save_files_on_server(files)
+    def self.save_files_on_server(files, resize_to_limit: nil)
```
Comment (madeindjs) [inline lib/active_storage/send_zip_helper.rb:16]: Can you add documentation about this params

~~~rb
# @param [Boolean] Allow resizing of files before downloading
~~~

File: lib/active_storage/send_zip_helper.rb [hunk 2]
```diff
@@ -81,7 +81,16 @@ def self.save_file_on_server(file, folder, subfolder: nil)
         filepath = File.join folder, filename
       end
 
-      File.open(filepath, 'wb') { |f| f.write(file.download) }
+      unless resize_to_limit.nil?
+        file = file.variant(resize_to_limit: resize_to_limit).processed
+      end
+
+      # read file from service
+      # contents = URI.open(file.service.send(:url_for, file.key)) { |f| f.read }
```
Comment (madeindjs) [inline lib/active_storage/send_zip_helper.rb:89]: Could you remove dead code ?

```suggestion
```

File: lib/active_storage/send_zip_helper.rb [hunk 3]
```diff
@@ -81,7 +81,16 @@ def self.save_file_on_server(file, folder, subfolder: nil)
...
+
+      # write the file on disk
+      File.open(filepath, 'wb') { |f| f.write(file.service.download(file.key)) }
```
Comment (madeindjs) [inline lib/active_storage/send_zip_helper.rb:92]: Is there any difference between `file.download` & `file.service.download(file.key)` ?

File: lib/active_storage/send_zip_helper.rb (hunk 1 above)
Comment (biximilien) [reply lib/active_storage/send_zip_helper.rb:16]: @madeindjs hi! Yes, I will make the requested changes.

File: lib/active_storage/send_zip_helper.rb (hunk 3 above)
Comment (biximilien) [reply lib/active_storage/send_zip_helper.rb:92]: @madeindjs #download is not available on ActiveStorage::Variant, however Service#download is available on both.

**Comment by madeindjs** [comment]
@biximilien , new version pushed https://rubygems.org/gems/active_storage-send_zip/versions/0.3.4 :tada:

**Pull request merged**